src/Npgsql/TypeHandlers/UnknownTypeHandler.cs

```diff
@@ -95,12 +95,22 @@ protected internal override Task WriteObjectWithLength(object value, NpgsqlWrite
             if (value == null || value is DBNull)
                 return base.WriteObjectWithLength(value, buf, lengthCache, parameter, async);
 
+            var convertedValue = value is string asString
+                ? asString
+                : (string)parameter.ConvertedValue;
```

Comment:
_If you wanted to_ you could probably rename `asString` to `s` and inline the whole ternary.